hazelcast/src/test/java/com/hazelcast/test/AbstractHazelcastClassRunner.java

```diff
@@ -409,6 +419,14 @@ public void evaluate() throws Throwable {
         private boolean isJUnitAssumeException(Throwable e) {
             return e instanceof AssumptionViolatedException;
         }
+
+        private boolean expectsException() throws IllegalAccessException {
+            if (expectedException == null) {
+                return false;
+            }
+            Object exceptionMatcher = ReflectionUtils.getFieldValueReflectively(expectedException, "matcherBuilder");
```

Comment:
The `ExpectedException` does not provide public API to indicate whether it was setup to expect an exception. This is achieved by obtaining the private `ExpectedExceptionMatcherBuilder matcherBuilder` and using the accessor below.